app/src/androidTest/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtilsTest.kt

```diff
@@ -107,6 +109,95 @@ class FirebaseDatabaseUtilsTest {
+        val dayOfMonth = mockNumberSnapshot(ldt.dayOfMonth)
+        `when`(snapshot.child("dayOfMonth")).thenReturn(dayOfMonth)
+        val hour = mockNumberSnapshot(ldt.hour)
+        `when`(snapshot.child("hour")).thenReturn(hour)
+        val minute = mockNumberSnapshot(ldt.minute)
+        `when`(snapshot.child("minute")).thenReturn(minute)
+        val second = mockNumberSnapshot(ldt.second)
+        `when`(snapshot.child("second")).thenReturn(second)
+        val nano = mockNumberSnapshot(ldt.nano)
+        `when`(snapshot.child("nano")).thenReturn(nano)
+
+        return snapshot
+    }
+
+    private fun mockTournamentPost(post: TournamentPost): DataSnapshot {
+        val snapshot = mock(DataSnapshot::class.java)
+
+        val userId = mockSnapshot(post.userId)
+        `when`(snapshot.child("userId")).thenReturn(userId)
+        val run = mockRun(post.run)
+        `when`(snapshot.child("run")).thenReturn(run)
+        val votes = mockNumberSnapshot(post.getVotes())
+        `when`(snapshot.child("votes")).thenReturn(votes)
+        val date = mockLocalDateTime(post.date)
+        `when`(snapshot.child("date")).thenReturn(date)
+        val usersVotes = mockSnapshot(post.getUsersVotes())
+        `when`(snapshot.child("usersVotes")).thenReturn(usersVotes)
+
+        return snapshot
+    }
+
+    private fun mockTournamentPostList(posts: List<TournamentPost>): DataSnapshot {
+        val snapshot = mock(DataSnapshot::class.java)
+
+        val postsSnap = posts.map {
+            mockTournamentPost(it)
+        }
+        `when`(snapshot.children).thenReturn(postsSnap)
+
+        return snapshot
+    }
+
+    private fun mockTournament(tournament: Tournament): DataSnapshot {
+        val snapshot = mock(DataSnapshot::class.java)
+
+        val id = mockSnapshot(tournament.id)
+        `when`(snapshot.child("id")).thenReturn(id)
+
+        val name = mockSnapshot(tournament.name)
+        `when`(snapshot.child("name")).thenReturn(name)
+
+        val description = mockSnapshot(tournament.description)
+        `when`(snapshot.child("description")).thenReturn(description)
+
+        val creatorId = mockSnapshot(tournament.creatorId)
+        `when`(snapshot.child("creatorId")).thenReturn(creatorId)
+
+        val startDate = mockLocalDateTime(tournament.startDate)
+        `when`(snapshot.child("startDate")).thenReturn(startDate)
+
+        val endDate = mockLocalDateTime(tournament.endDate)
+        `when`(snapshot.child("endDate")).thenReturn(endDate)
+
+        val participants = mock(DataSnapshot::class.java)
+        val participantsSubNodes = tournament.participants.map {
+            val participant = mock(DataSnapshot::class.java)
+            `when`(participant.value).thenReturn(true)
+            `when`(participant.key).thenReturn(it)
+            participant
+        }
+        `when`(snapshot.child("participants")).thenReturn(participants)
+        `when`(participants.children).thenReturn(participantsSubNodes)
```

Comment:
Participants is just a mock `DataSnapshot,` it does not contain any data, it basically just reacts to some calls (the ones in the `when` clauses). So here the only thing happenning with `participants` is to return a list of Snapshots when `partcipants.children` is called. 

It is implemented like that because for now in the real database, only `children` is called on `participants`, but if we need to use other properties like `child` or `key` we should add additionnal `when` clauses.

Reply:
Oh i see now, thanks for the explaination !